src/frontend/Semantic_error.ml

```diff
@@ -152,6 +156,68 @@ module TypeError = struct
           (String.concat ~sep:"" lines)
           Fmt.(list UnsizedType.pp ~sep:comma)
           arg_tys
+    | IllTypedVariadicODE (name, arg_tys, args) ->
```

Comment:
I think this could still have been two errors like you had it, I only meant to suggest that they could both use a helper function because their logic overlapped so much. But this works too, I'm sorry if I put you through extra effort

Reply:
No problem, I think this is more concise and also makes less code to maintain or fix when we add the type for variadic functions.